lib/launchEditor.js

```diff
@@ -0,0 +1,307 @@
+ * @type {childProcess.ChildProcess | null}
+ */
+let _childProcess = null;
+
+/**
+ *
+ * @param {ReturnType<Compiler["getInfrastructureLogger"]>} logger
+ * @param {string} fileName
+ * @returns {void}
+ */
+function launchEditor(logger, fileName) {
+  if (!fs.existsSync(fileName)) {
+    return;
+  }
+
+  // eslint-disable-next-line prefer-const
+  let [editor, ...args] = guessEditor();
+
+  if (!editor) {
+    printInstructions(logger, fileName, null);
+    return;
+  }
+
+  if (editor.toLowerCase() === "none") {
+    return;
+  }
+
+  if (
+    process.platform === "linux" &&
+    fileName.startsWith("/mnt/") &&
+    /Microsoft/i.test(os.release())
+  ) {
+    // Assume WSL / "Bash on Ubuntu on Windows" is being used, and
+    // that the file exists on the Windows file system.
+    // `os.release()` is "4.4.0-43-Microsoft" in the current release
+    // build of WSL, see: https://github.com/Microsoft/BashOnWindows/issues/423#issuecomment-221627364
+    // When a Windows editor is specified, interop functionality can
+    // handle the path translation, but only if a relative path is used.
+    fileName = path.relative("", fileName);
+  }
+
+  // cmd.exe on Windows is vulnerable to RCE attacks given a file name of the
+  // form "C:\Users\myusername\Downloads\& curl 172.21.93.52". Use a whitelist
+  // to validate user-provided file names. This doesn't cover the entire range
+  // of valid file names but should cover almost all of them in practice.
+  if (
+    process.platform === "win32" &&
+    !WINDOWS_FILE_NAME_WHITELIST.test(fileName.trim())
+  ) {
+    logger.error(`Could not open ${path.basename(fileName)} in the editor.`);
+    logger.info(`When running on Windows, file names are checked against a whitelist to protect
+    against remote code execution attacks. File names may consist only of alphanumeric characters 
+    (all languages), periods, dashes, slashes, and underscores.`);
+    return;
+  }
+
+  args.push(fileName);
+
+  if (_childProcess && isTerminalEditor(editor)) {
+    // There's an existing editor process already and it's attached
+    // to the terminal, so go kill it. Otherwise two separate editor
+    // instances attach to the stdin/stdout which gets confusing.
+    _childProcess.kill("SIGKILL");
+  }
+
+  if (process.platform === "win32") {
+    // On Windows, launch the editor in a shell because spawn can only
+    // launch .exe files.
+    _childProcess = childProcess.spawn("cmd.exe", ["/C", editor].concat(args), {
+      stdio: "inherit",
+    });
+  } else {
+    _childProcess = childProcess.spawn(editor, args, { stdio: "inherit" });
+  }
+
+  if (_childProcess) {
+    _childProcess.on("exit", (errorCode) => {
+      _childProcess = null;
+
+      if (errorCode) {
+        printInstructions(logger, fileName, `(code ${errorCode})`);
+      }
+    });
+
+    _childProcess.on("error", (error) => {
+      printInstructions(logger, fileName, error.message);
+    });
+  }
+}
+
+module.exports = launchEditor;
```

Comment:
Maybe there is a package for this? Also can we add unit tests for this?

Reply:
I've installed [`launch-editor`](https://github.com/yyx990803/launch-editor) package, which apparently is extracted from `react-dev-utils`. It's also used by Vite as well.